rules/framework/vfs_overlay.bzl

```diff
@@ -43,6 +43,20 @@ def _make_root(ctx, framework_name, root_dir, extra_search_paths, module_map, hd
             "external-contents": module_map[0].path,
         })
 
+    if len(swiftmodule):
+        modules_contents.append({
+            "type": "directory",
+            "name": swiftmodule[0].dirname.split("/").pop(),
```

Comment:
A few notes here:
only works if name is just the ABC.xcframework without path. This makes sense because `name` here is appended to the current transverse of the json tree (let me know if u don't understand this)
each file inside needs to be explicit, saying swiftmodule is a file as a while does not work.
Let me know any above is incorrect

Reply:
Yep this looks like it might have some issues generalizing. Check my other comment on the `SwiftInfo` import - I was wondering if we'd get the imported framework's swiftmodule into the provider already and if we just needed to add it to `deps`